IONFileViewerLib/IONFLVWManager.swift

```diff
@@ -1,3 +1,110 @@
-class IONFLVWManager {
-    // TODO add implementation
+import Foundation
+import UIKit
+
+public class IONFLVWManager {
+    private let fileManager: FileManager
+    weak var rootViewController: UIViewController?
+
+    public init(viewController: UIViewController, fileManager: FileManager = .default) {
+        self.rootViewController = viewController
+        self.fileManager = fileManager
+    }
+}
+
+extension IONFLVWManager: IONFLVWOpenDocumentManager {
+    public func openDocumentFromLocalPath(filePath: String, completion: @escaping (() -> Void)) throws {
+        guard !filePath.isEmpty, let url = filePath.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) else {
+            throw IONFLVWError.emptyFilePath
+        }
+        guard let file = URL(string: url) else { throw IONFLVWError.couldNotOpenDocument }
+        guard file.lastPathComponent.contains(".") else { throw IONFLVWError.missingFileExtension }
```

Comment:
We could use `file.pathExtension.isEmpty == false` instead which I believe may be more reliable.